zookeeper-api/src/main/java/org/apache/helix/zookeeper/api/client/HelixZkClient.java

```diff
@@ -89,32 +88,17 @@ public int getSessionTimeout() {
    * Deprecated - please use RealmAwareZkClient and RealmAwareZkClientConfig instead.
    *
    * Configuration for creating a new HelixZkClient with serializer and monitor.
-   *
-   * TODO: If possible, try to merge with RealmAwareZkClient's RealmAwareZkClientConfig to reduce duplicate logic/code (without breaking backward-compatibility).
-   * Simply making this a subclass of RealmAwareZkClientConfig will break backward-compatiblity.
    */
   @Deprecated
-  class ZkClientConfig {
-    // For client to init the connection
-    private long _connectInitTimeout = DEFAULT_CONNECTION_TIMEOUT;
-
-    // Data access configs
-    private long _operationRetryTimeout = DEFAULT_OPERATION_TIMEOUT;
-
-    // Others
-    private PathBasedZkSerializer _zkSerializer;
-
-    // Monitoring
-    private String _monitorType;
-    private String _monitorKey;
-    private String _monitorInstanceName = null;
-    private boolean _monitorRootPathOnly = true;
+  class ZkClientConfig extends RealmAwareZkClientConfig {
 
+    @Override
```

Comment:
Why do we need to re-implement these methods if you inherit from the RealmAwareZkClientConfig?

Reply:
As I explained, the issue was trying to

> cast superclass to subclass which is not allowed

I had to override so that these setters return an instance of HelixZkClient.ZkClientConfig instead of RealmAwareZkClientConfig. I suggest thinking more carefully about the return types and their inheritance relationships.